parser_library/test/workspace/pathmask_test.cpp

```diff
@@ -186,6 +184,16 @@ TEST(pathmask, fail)
     EXPECT_FALSE(check_path("/path/?*?*?/test/", "/path/a/b/test/"));
     EXPECT_FALSE(check_path("/path/?**?/test/", "/path/a//test/"));
 
+    EXPECT_FALSE(check_path("/path/?/test/", "/path/%7d/test/")); // lowercase percent encoding is not allowed
+    EXPECT_FALSE(check_path("/path/?/test/", "/path/%7D%7D/test/"));
+    EXPECT_FALSE(check_path("/path/?/test/", "/path/%7D%DF%BF/test/"));
+    EXPECT_FALSE(check_path("/path/?/test/", "/path/%DF%BF%7D/test/"));
+
+    // %FF is not a valid UTF-8 character
+    EXPECT_FALSE(check_path("/path/?/test/", "/path/%FF/test/"));
+    EXPECT_FALSE(check_path("/path/?/test/", "/path/%EF%BF%FF/test/"));
+    EXPECT_FALSE(check_path("/path/\?\?/test/", "/path/%EF%BF%FF/test/"));
+
```

Comment:
are these paths or urls? because I can create a folder called "%FF"

Reply:
It's supposed to be already UTF-8 encoded string (shouldn't matter whether this is a path or not). 

So maybe this file and `pathmask_to_regex()` should be called differently? Like `utf8_path_to_regex_test.cpp` and `utf8_path_to_regex()`?